packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -225,6 +225,15 @@ export class Cluster extends Resource implements ICluster {
       this.enableFargateCapacityProviders();
     }
 
+    if (this._capacityProviderNames.length > 0 && props.defaultCapacityProviderStrategy && props.defaultCapacityProviderStrategy.length > 0) {
```

Comment:
I don't know if you need this check for the length of defaultCapacityProviderStrategy; you should be able to just check existence and rely on forEach to iterate 0 times for a 0-length array.

```suggestion
    if (this._capacityProviderNames.length > 0 && props.defaultCapacityProviderStrategy) {
```